### DISCUSSION TITLE
Is it possible to make just some of tha APIs available to the public?

### DISCUSSION THREAD

**Discussion by junoriosity** [fastapi/fastapi#4642] [Questions] [answered]
I want to run my FastAPI server in Kubernetes and expose the APIs.

There are two routes in `main.py`

```
...
app.include_router(projects.router, prefix="/private", tags=["private"])
app.include_router(projects.router, prefix="/public", tags=["public"])
```

Now I want to expose that service to the public, say, via a reverse proxy or via a LoadBalancer. Now I just want to make the `public` routing available there. Within the Kubernetes cluster, however, both the `private` and `public` routes should be reachable.

Is that possible and if so, how?

**Comment by vanishingESCkey** [comment, marked as answer]
You may consider to set this limitation using your proxy or load balancer, so out of fastapi app.
Here is nginx setup used in my project where /api/internal is available only for limited IPs including localhost.
```
upstream appx {
  server 127.0.0.1:3003;
}

server {

  # ... other configuration

  location ^~ /api/internal {
    proxy_pass http://appx;
    # ... other proxy directives
    allow 127.0.0.1;        # localhost
    # ... other allow ips
    deny all;
  }


  location / {
    proxy_pass http://appx;
    # ... other proxy directives
  }

}
```
